radix-engine/src/errors.rs

```diff
@@ -411,7 +415,15 @@ pub enum ApplicationError {
     // TODO: this should never happen because of schema check?
     InputDecodeError(DecodeError),
 
-    Panic(String),
+    /// A panic was encountered in Scrypto code.
+    ScryptoBlueprintExecutionPanic(String),
+
+    /// A panic was encountered in Native code.
+    NativeBlueprintExecutionPanic {
+        export_name: String,
+        input: ScryptoValue,
+        error: String,
+    },
```

Comment:
Can we revert the changes to the `Panic(String)`, which is intended for emitting message **before abortion**.

Instead, suggest to have `NativeRuntimeError::Trap(String)`